softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java

```diff
@@ -1,20 +1,33 @@
 package com.softgroup.common.router.api;
 
 
+import com.softgroup.common.datamapper.DataMapper;
+import com.softgroup.common.datamapper.JacksonDataMapper;
 import com.softgroup.common.protocol.Request;
 import com.softgroup.common.protocol.RequestData;
 import com.softgroup.common.protocol.Response;
 import com.softgroup.common.protocol.ResponseData;
 
+import java.util.Map;
+
 public abstract class AbstractRequestHandler<T extends RequestData, R extends ResponseData> implements RequestHandler {
-	@Override
+
+	DataMapper mapper = new JacksonDataMapper();
+
 	public String getName() {
 		return null;
 	}
 
+	public abstract Response<R> doHandle(Request<T> msg);
+
 	@Override
 	public Response<R> handle(Request<?> msg) {
-		return null;
+		Map<String, Object> map = mapper.convertToMap(msg);
```

Comment:
I dont know how to convert Request<?> to Request<T> using DataMapper one time. I need a hint.